Modules/Forum/classes/class.ilObjForumGUI.php

```diff
@@ -2967,7 +2967,9 @@ public function viewThreadObject(): void
             ->current();
 
         $additionalDataExists = $toolContext->getAdditionalData()->exists(ForumGlobalScreenToolsProvider::SHOW_FORUM_THREADS_TOOL);
-        if (!$additionalDataExists && $this->selectedSorting === ilForumProperties::VIEW_TREE) {
+        $subtree_nodes = $this->objCurrentTopic->getPostTree($firstNodeInThread);
```

Comment:
1. If we add more checks before we display the tree in the tool slate, we should move this below the access control check (`ensureThreadBelongsToForum` IMO.
2. If you move the retrieval of the nodes, you also have to move `$this->objCurrentTopic->setOrderField($orderField);`. This means you have to move even more, because `$orderField` (should be renamed to `$order_field`) to be move as well (and so on) :-).